src/Backend/SgateShopgatePlugin/Plugin.php

```diff
@@ -4424,6 +4424,38 @@ public function checkCart(ShopgateCart $cart)
         Shopware()->Session()->sUserGroup     = $groupData['id'];
         Shopware()->Session()->sUserGroupData = $groupData;
 
+        $userDeliveryAddress = $cart->getDeliveryAddress();
+        if (!empty($userDeliveryAddress)
+            && ($userCountry = $userDeliveryAddress->getCountry())
+            // assignment on purpose to make sure the same country is used in every place
+            && strlen($userCountry) > 0
+        ) {
+            $country       = $this->customerImport->getCountryByIso($userDeliveryAddress->getCountry());
+            $state         = $this->customerImport->getStateByIso($userDeliveryAddress->getState());
+            $userCountryId = $country->getId();
+            $userAreaId    = $country->getArea()->getId();
+            $userStateId   = $state ? $state->getId() : 0;
+
+            if (Shopware()->Session()->sCountry != $userCountryId
+                || Shopware()->Session()->sState != $userStateId
+                || Shopware()->Session()->sArea != $userAreaId
+            ) {
+                Shopware()->Session()->sCountry = $userCountryId;
+                Shopware()->Session()->sState   = $userStateId;
+                Shopware()->Session()->sArea    = $userAreaId;
+                Shopware()->Container()->get('shopware_storefront.context_service')->initializeShopContext();
```

Comment:
This method was introduced with Shopware 5.x. Please check back with a PO if we still should support 4.x as we are also using the context service in a different place of our plugin.